test/minikube/001-kube.bats

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bats
+#
+# Tests of podman kube commands with minikube
+#
+
+load ../system/helpers.bash
+
+function setup(){
+    # only set up the minikube cluster before the first test
+    if [ "$BATS_TEST_NUMBER" -eq 1 ]; then
+        minikube start
+        echo $output >&2
+    fi
+    # wait for a bit so all the services in the cluster come up
+    sleep 10
+}
+
+###############################################################################
+# BEGIN tests
+
+@test "minikube - check cluster is up" {
+    run minikube kubectl get nodes
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"Ready"* ]]
+    run minikube kubectl get pods
+    [ "$status" -eq 0 ]
+    [[ "$output" == "No resources found in default namespace." ]]
+}
+
+@test "minikube - deploy generated container yaml to minikube" {
+    cname="test-ctr"
+    fname="/tmp/f$(random_string 6).yaml"
+    run_podman container create --name $cname $IMAGE top
+    [ "$status" -eq 0 ]
+    run_podman kube generate -f $fname $cname
+    [ "$status" -eq 0 ]
+
+    # deploy to the minikube cluster
+    run minikube kubectl -- apply -f $fname
+    [ "$status" -eq 0 ]
+    [[ "$output" == "pod/$cname-pod created" ]]
+    # need to wait for the pod to be created and started before checking status
+    sleep 5
```

Comment:
Non-synchronize-able actions like `sleep` are almost always a bad idea to use in tests since there is very little control over the operating conditions at this level.  I highly recommend using some kind of timeout-wait loop to synchronize against some known/expected state.

Reply:
Fixed